morebits.js

```diff
@@ -2928,11 +2939,23 @@ Morebits.wiki.page = function(pageName, currentAction) {
 		if ($(xml).find('page').attr('title')) {
 			var resolvedName = $(xml).find('page').attr('title');
 
-			// only notify user for redirects, not normalization
 			if ($(xml).find('redirects').length > 0) {
+				// check for cross-namespace redirect:
+				var origNs = new mw.Title(ctx.pageName).namespace;
+				var newNs = new mw.Title(resolvedName).namespace;
+				if (origNs !== newNs && !ctx.followCrossNsRedirect) {
+					ctx.statusElement.error(ctx.pageName + ' is a cross-namespace redirect to ' + resolvedName);
+					return;
+				}
+
+				// only notify user for redirects, not normalization
 				Morebits.status.info('Info', 'Redirected from ' + ctx.pageName + ' to ' + resolvedName);
+				ctx.pageName = resolvedName; // update to redirect target
+
+			} else {
+				ctx.pageName = resolvedName;  // update in case of normalization
 			}
-			ctx.pageName = resolvedName;  // always update in case of normalization
```

Comment:
Couldn't this just stay here instead of being repeated above?  I suppose it's clearer this way.

Reply:
Yes it could. Removed the repetition.